megabus/src/main/java/com/bazaarvoice/megabus/refproducer/MegabusRefProducerManager.java

```diff
@@ -53,26 +61,35 @@ public MegabusRefProducerManager(final LifeCycleRegistry lifeCycle,
         // servers.
-
-        // TODO: since partitioned databus subscriptions are 1-based, we must add one to the partition condition. At some point in the future,
-        // we should reconcile this inconsistency
-        PartitionedServiceSupplier refProducerSupplier = partition ->
-                new MegabusRefProducer(refProducerConfiguration, databusFactory.forOwner(systemId), databusEventStore,
-                        Conditions.partition(numRefPartitions, partition + 1),
-                        logFactory, metricRegistry, kafkaCluster.producer(), objectMapper, refTopic,
-                        Integer.toString(partition), applicationId);
-
-        PartitionedLeaderService partitionedLeaderService = new PartitionedLeaderService(
-                curator, LEADER_DIR, hostAndPort.toString(),
+        super(curator, LEADER_DIR, hostAndPort.toString(),
                 "PartitionedLeaderSelector-MegabusRefProducer", numRefPartitions, 1, 1, TimeUnit.MINUTES,
-                refProducerSupplier, clock);
+                partition -> new MegabusRefProducer(refProducerConfiguration, databusEventStore,
+                                logFactory, metricRegistry, kafkaCluster.producer(), objectMapper, refTopic,
+                                ChannelNames.getMegabusRefProducerChannel(applicationId, partition), Integer.toString(partition)), clock);
+
+        _databus = checkNotNull(databusFactory).forOwner(systemId);
+        _numRefPartitions = numRefPartitions;
+        _applicationId = checkNotNull(applicationId);
 
-        for (LeaderService leaderService : partitionedLeaderService.getPartitionLeaderServices()) {
+        for (LeaderService leaderService : getPartitionLeaderServices()) {
             ServiceFailureListener.listenTo(leaderService, metricRegistry);
         }
 
-        leaderServiceTask.register("megabus-ref-producer", partitionedLeaderService);
-        lifeCycle.manage(partitionedLeaderService);
+        leaderServiceTask.register("megabus-ref-producer", this);
+    }
+
+    public void createRefSubscriptions() {
+        // TODO: since partitioned databus subscriptions are 1-based, we must add one to the partition condition. At some point in the future,
+        // we should reconcile this inconsistency
+
+        // Except for resetting the ttl, recreating a subscription that already exists has no effect.
+        // Assume that multiple servers that manage the same subscriptions can each attempt to create
+        // the subscription at startup.  The subscription should last basically forever.
+        for (int i = 0; i < _numRefPartitions; i++) {
+            _databus.subscribe(ChannelNames.getMegabusRefProducerChannel(_applicationId, i),
+                    Conditions.partition(_numRefPartitions, i + 1),
+                    DatabusChannelConfiguration.MEGABUS_TTL, DatabusChannelConfiguration.MEGABUS_TTL, false);
```

Comment:
What kind of TTL's were you thinking? If you mean that the subscription and event TTL's should be different, I don't think it matters here, as both need to be effectively forever.

Reply:
yeah, was thinking about subscription and event TTLs, so we won't accidentally change one of them as it was during outage.

Reply:
In regard to the `_databus.subscribe` deprecation, I chose the deprecated method as matched the canary, but I looking at it now, I see no reason it can't be changed.

Reply:
Actually I'm wrong, we can't change this. The reason that the canary, and also the megabus, cannot use the non-deprecated method is that they require that `includeDefaultJoinFilter` be false because they to get _all_ updates, regardless of their source. The default join filter was created a source of protection for Polloi from Re-ETL's. the megabus and it's consumers shouldn't need such protection.

Reply:
in this case why method is marked deprecated, but we are actively using it?

Reply:
That's a very good question, and we might have to some software archaeology to find out.